### DISCUSSION TITLE
Support for pytest

### DISCUSSION THREAD

**Discussion by fsuba** [teemtee/tmt#2159] [Ideas]
Currently there are two frameworks supported in tmt tests specification:
```
framework: shell
framework: beakerlib
```
Wouldn't it make sense to also include pytest?
Also, plan prepare step accepts package, which has to be rpm. What if we could also specify pypi packages to be installed?
When pytest framework is selected, there could be python venv created for that plan.
Related to https://github.com/teemtee/tmt/discussions/1922

**Comment by happz** [comment]
There were some voices looking for this kind of support, and there's an issue for leveling the playfield to allow other players besides `shell` and `beakerlib`: https://github.com/teemtee/tmt/issues/1305

**Comment by thrix** [comment]
Btw, people now run `pytest` tests, see convert2rhel, e.g. https://github.com/oamg/convert2rhel/tree/main/tests/integration/tier0/assessment-report

Of course, tighter integration could be nicer, usability wise, but I believe you can now run `pytest` tests via `tmt` ...

Patches welcome ;) `tmt` is open-source.
